### PR TITLE
Itk: 5.0.1 -> 5.1.0

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build elastix ezminc simpleitk vtk itk4

**Comment by veprbl** [comment]
@GrahamcOfBorg build elastix ezminc simpleitk vtk itk4

File: pkgs/development/libraries/science/biology/elastix/default.nix [hunk 1]
```diff
@@ -11,10 +11,24 @@ stdenv.mkDerivation rec {
...
+      sha256 = "1hf7kgx1jv497pf0x5wj79sy1wncxcvhrkix9w086lcr8zwxvn9q";
+    })
+  ];
+
+
   nativeBuildInputs = [ cmake python3 ];
   buildInputs = [ itk ];
 
-  doCheck = true;
+  doCheck = !stdenv.isDarwin;  # usual dynamic linker issues
```
Comment (veprbl) [inline pkgs/development/libraries/science/biology/elastix/default.nix:31]: What do you mean by usual issues? Can we resolve with `preCheck = "export DYLD_LIBRARY_PATH ...`?

**Comment by veprbl** [comment]
@GrahamcOfBorg build elastix ezminc simpleitk vtk itk4

File: pkgs/development/libraries/science/biology/elastix/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/libraries/science/biology/elastix/default.nix:31]: I tried doing something naive similar to what is done with LD_LIBRARY_PATH already but that didn't work.  I don't really do much development on OS X so don't mind disabling this package if you prefer.

**Review by jonringer — COMMENTED** [review]
looks like h5c++ want's pkg-config to be available at runtime:
```
/nix/store/qvb7wnswcaqspp7k6v8ak18llfx89gw0-itk-5.1.0/h5c++
dir is /nix/store/qvb7wnswcaqspp7k6v8ak18llfx89gw0-itk-5.1.0
./results/itk/bin/h5c++: line 32: pkg-config: command not found
```

**Comment by bcdarwin** [comment]
@jonringer -- `pkgconfig` now propagated.

**Comment by knedlsepp** [comment]
> @jonringer -- `pkgconfig` now propagated.

Hm. This means it will only be available to all nix-builds depending on it, but not at runtime when using `nix run nixpkgs.itk`. If we want the latter we probably will need to add the pkg-config PATH to the makeWrapperArgs, right?

**Comment by bcdarwin** [comment]
@knedlsepp -- h5cc and h5c++ are compiler wrappers so most likely they would be used by other derivations, although it's a bit unseemly to have them fail at runtime.  Not sure what the best solution is here -- these are somewhat incidental programs built by the ITKHDF5 module which aren't needed to compile most programs using ITK (including all reverse deps in the Nixpkgs tree such as ANTs).  Is it worth it to create a wrapper for them?

**Comment by bcdarwin** [comment]
I've wrapped h5c++ with pkgconfig.

File: pkgs/development/libraries/itk/default.nix [hunk 2]
```diff
@@ -23,12 +24,17 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ cmake xz ];
-  buildInputs = [ libX11 libuuid vtk ] ++ stdenv.lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Cocoa ];
+  nativeBuildInputs = [ cmake xz makeWrapper ];
+  buildInputs = [ libX11 libuuid vtk ] ++ stdenv.lib.optionals stdenv.isDarwin [ Cocoa ];
+  propagatedBuildInputs = [ pkgconfig ];
+
+  postInstall = with stdenv.lib; ''
```
Comment (veprbl) [inline pkgs/development/libraries/itk/default.nix:31]:
```suggestion
  postInstall = ''
```

**Comment by veprbl** [comment]
@GrahamcOfBorg build elastix ezminc simpleitk vtk itk4

File: pkgs/development/libraries/itk/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/libraries/itk/default.nix:31]: Done, thanks.

**Comment by bcdarwin** [comment]
@GrahamcOfBorg build vtk itk elastix ezminc simpleitk

**Comment by veprbl** [comment]
@GrahamcOfBorg build elastix ezminc simpleitk vtk itk4

File: pkgs/development/libraries/itk/default.nix [hunk 3]
```diff
@@ -23,12 +24,17 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ cmake xz ];
-  buildInputs = [ libX11 libuuid vtk ] ++ stdenv.lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Cocoa ];
+  nativeBuildInputs = [ cmake xz makeWrapper ];
+  buildInputs = [ libX11 libuuid vtk ] ++ stdenv.lib.optionals stdenv.isDarwin [ Cocoa ];
+  propagatedBuildInputs = [ pkgconfig ];
```
Comment (veprbl) [inline pkgs/development/libraries/itk/default.nix:29]: I guess this one is not needed anymore. It also should have been `propagatedNativeBuildInputs`.

**Review by veprbl — APPROVED** [review]

**Pull request merged**